x-pack/plugins/enterprise_search/public/applications/app_search/components/result_settings/sample_response/sample_response_logic.ts

```diff
@@ -0,0 +1,98 @@
+
+const NO_RESULTS_MESSAGE = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.resultSettings.sampleResponse.noResultsMessage',
+  { defaultMessage: 'No results.' }
+);
+
+const ERROR_MESSAGE = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.resultSettings.sampleResponse.errorMessage',
+  { defaultMessage: 'An error occured.' }
+);
+
+interface SampleResponseValues {
+  query: string;
+  response: SampleSearchResponse | string | null;
+}
+
+interface SampleResponseActions {
+  queryChanged: (query: string) => { query: string };
+  getSearchResultsSuccess: (response?: SampleSearchResponse) => { response?: SampleSearchResponse };
+  getSearchResultsFailure: (response?: string) => { response?: string };
+  getSearchResults: (
+    query: string,
+    resultFields: ServerFieldResultSettingObject
+  ) => { query: string; resultFields: ServerFieldResultSettingObject };
+}
+
+export const SampleResponseLogic = kea<MakeLogicType<SampleResponseValues, SampleResponseActions>>({
+  path: ['enterprise_search', 'app_search', 'sample_response_logic'],
+  actions: {
+    queryChanged: (query) => ({ query }),
+    getSearchResultsSuccess: (response) => ({ response }),
+    getSearchResultsFailure: (response) => ({ response }),
+    getSearchResults: (query, resultFields) => ({ query, resultFields }),
+  },
+  reducers: {
+    query: ['', { queryChanged: (_, { query }) => query }],
+    response: [
+      null,
+      {
+        getSearchResultsSuccess: (_, { response }) => response || null,
+        getSearchResultsFailure: (_, { response }) => response || null,
+      },
+    ],
+  },
+  listeners: ({ actions }) => ({
+    getSearchResults: async ({ query, resultFields }, breakpoint) => {
+      if (Object.keys(resultFields).length < 1) return;
+      await breakpoint(250);
+
+      const { http } = HttpLogic.values;
+      const { engineName } = EngineLogic.values;
+
+      const url = `/api/app_search/engines/${engineName}/sample_response_search`;
+
+      try {
+        const response = await http.post(url, {
+          body: JSON.stringify({
+            query,
+            result_fields: resultFields,
+          }),
+        });
+
+        const result = response.results?.[0];
+        actions.getSearchResultsSuccess(
+          result ? { ...result, _meta: undefined } : NO_RESULTS_MESSAGE
+        );
+      } catch (e) {
+        if (e.response.status >= 500) {
+          // 4XX Validation errors are expected, as a user could enter something like 2 as a size, which is out of valid range.
+          // In this case, we simply render the message from the server as the response.
+          //
+          // 5xx Server errors are unexpected, and need to be reported in a flash message.
+          flashAPIErrors(e);
+          actions.getSearchResultsFailure(ERROR_MESSAGE);
+        } else {
+          actions.getSearchResultsFailure(e.body?.message || ERROR_MESSAGE);
```

Comment:
This is like, a slight difference between this UI and the old UI...

The old UI showed:
<img width="465" alt="error2" src="https://user-images.githubusercontent.com/1427475/113324857-992d9700-92e5-11eb-9caf-d14f11f70487.png">

This UI shows:
<img width="530" alt="error message" src="https://user-images.githubusercontent.com/1427475/113324742-7a2f0500-92e5-11eb-8465-6716e98fbf86.png">

I did it this way simply because for whatever reason, "e.body?.message" had this message readily available, I didn't have to parse the response or anything. I'm not exactly sure where that was parsed into a message ... maybe Kibana does that automatically 🤔 .

I could change it back to the way it was in the UI. I wasn't really sure which was better.

Reply:
Ya I think the `errors:` key is a clearer indication as well that it's a response from the API, if it's not too much dev effort I'd like to structure it as before. If it's more than half an hour of dev effort however I'd say leave it for now since Davey wants to come back to this UI and redesign it at some point anyway

Reply:
08737c4532726ef1cff4a17a40eb8e305972be1f

Reply:
<img width="1577" alt="whateever" src="https://user-images.githubusercontent.com/1427475/113591653-2df40580-9602-11eb-95ad-1c9eeca84591.png">